src/town_cmd.cpp

```diff
@@ -1087,6 +1087,55 @@ static bool GrowTownWithRoad(const Town *t, TileIndex tile, RoadBits rcmd)
 	return false;
 }
 
+/**
+ * Checks if a town road can be continued on the other side of a bridge.
+ *
+ * @param end_tile The end tile of the bridge
+ * @param bridge_dir The direction of the bridge
+ * @return true if the road can be continued, else false
+ */
+static bool CanRoadContinueAfterBridge(const Town* t, const TileIndex end_tile, const DiagDirection bridge_dir)
+{
+	const int delta = TileOffsByDiagDir(bridge_dir); // +1 tile in the direction of the bridge
+	TileIndex next_tile = end_tile + delta; // The tile beyond the bridge
+	RoadBits rcmd = DiagDirToRoadBits(ReverseDiagDir(bridge_dir)); // The road bits that would be built after extending the bridge.
+	RoadType rt = GetTownRoadType(t); // The road type being built by the town.
+
+	/* Before we try anything, make sure the tile is on the map and not the void. */
+	if (!IsValidTile(next_tile)) return false;
+
+	/* If the next tile is a bridge or tunnel, allow if it's a road bridge/tunnel continuing in the same direction. */
+	if (IsTileType(next_tile, MP_TUNNELBRIDGE)) {
+		if (GetTunnelBridgeTransportType(next_tile) == TRANSPORT_ROAD && GetTunnelBridgeDirection(next_tile) == bridge_dir) return true;
+	}
+
+	/* If the next tile is a station, allow if it's a road station facing the proper direction. Otherwise return false. */
+	if (IsTileType(next_tile, MP_STATION)) {
+		/* If the next tile is a road station, allow if it's facing the same direction, otherwise disallow. */
+		if (IsRoadStop(next_tile)) {
+			if (GetRoadStopDir(next_tile) == ReverseDiagDir(bridge_dir)) return true;
+
+			return false; // Tile is a station, but not a road station.
+		}
```

Comment:
Maybe?
```suggestion
		return IsRoadStop(next_tile) && GetRoadStopDir(next_tile) == ReverseDiagDir(bridge_dir);
```

Reply:
According to the docs, && in C++ is short-circuiting so I believe this would work. If `GetRoadStopDir()` is called on a non-road-stop tile, it returns an assertion error which crashes the game. What is the best practice here?

Reply:
`GetRoadStopDir()` won't be called if `IsRoadStop()` fails, so it's ok.
You can see similar constructs in station_map.h (and many other locations)

Reply:
Thanks. I'll do that in the next round of revisions.